accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/rest/dto/HistoryEventDTO.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.commons.accession.rest.dto;
+
+import uk.ac.ebi.ampt2d.commons.accession.core.models.EventType;
+import uk.ac.ebi.ampt2d.commons.accession.core.models.HistoryEvent;
+
+import java.time.LocalDateTime;
+import java.util.function.Function;
+
+public class HistoryEventDTO<ACCESSION, DTO> {
+
+    private EventType type;
+
+    private ACCESSION accession;
+
+    private Integer version;
+
+    private ACCESSION mergedInto;
+
+    private LocalDateTime localDateTime;
+
+    private DTO data;
+
+    public <MODEL> HistoryEventDTO(HistoryEvent<MODEL, ACCESSION> event, Function<MODEL, DTO> modelToDTO) {
+        this.type = event.getType();
+        this.accession = event.getAccession();
+        this.mergedInto = event.getMergedInto();
+        this.localDateTime = event.getLocalDateTime();
+        this.data = modelToDTO.apply(event.getData());
```

Comment:
The conversion from entity to model now works differently than the conversion from model to DTO. This one uses a `Function` whereas the other is a direct cast.

Reply:
We can't do a direct cast for the DTO. We need to use the DTO object if we want to use them to define the mapping for the json objects.